enhancements/storage/csi-driver-install.md

```diff
@@ -0,0 +1,149 @@
+---
+title: CSI driver installation for in-tree drivers
+authors:
+  - "@gnufied"
+  - “@jsafrane”
+reviewers:
+  - "@eparis”
+approvers:
+  - TBD
+  - "@..."
+creation-date: 2019-11-04
+last-updated: 2019-11-04
+status: provisional
+see-also:
+replaces:
+superseded-by:
+---
+
+# Installation of CSI drivers that replace in-tree drivers
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/**
+
+## Summary
+
+We want certain CSI drivers such as AWS, GCE, Cinder, Azure and vSphere to be installable on Openshift, so as
+they can be used along-side in-tree drivers and when upstream enables migration flag for these volume types, their
+replacement CSI drivers can take over and none of the storage features get affected.
+
+## Motivation
+
+Upstream Kubernetes is moving towards removing code of in-tree drivers and replacing them with their CSI counterpart. Our
+current expectation is that - all in-tree drivers that depend on cloudprovider should be removed from core Kubernetes by 1.21.
+This may not happen all at once and we expect migration for certain drivers to happen sooner.
+
+This does mean that - Openshift should be prepared to handle such migration. We have to iron out any bugs in driver themselves and
+their interfacing with Openshift. We need a way for users to use the CSI drivers and optionally enable migration from in-tree driver
+to CSI driver. To support upstream design - we will also need a way for users to disable the migration and keep using in-tree driver, until
+in-tree code is finally removed.
+
+## Goals
+
+We would like to approach various goals of this KEP in different phases, because we do not expect to get everything right or done in phase-1.
+
+### Phase-1 Goals
+
+* Provide a way for AWS CSI driver installation starting from Openshift-4.4. Users could either optionally install it or CSI driver could be installed
```

Comment:
I'm against supporting both at once.  If we have the out of tree provider and we're ready to support it, we should make a clean cut-over to that.  We need to keep our supportability matrix small, and it's much easier to do a migration when you know you go from 4.x -> 4.x+1, rather than having to figure out how and what the user deployed.